code/messaging/src/phone/java/com/adobe/marketing/mobile/InAppNotificationHandler.java

```diff
@@ -49,204 +57,89 @@ class InAppNotificationHandler {
-            }
-        } else { // otherwise use the application identifier to validate offers
-            Log.trace(LOG_TAG, "%s - Using the application identifier (%s) to validate offers.", SELF_TAG, offersConfig.applicationId);
-            final EncodingService encodingService = MobileCore.getCore().eventHub.getPlatformServices().getEncodingService();
-            String decisionScope;
-            try {
-                Object rawScope = payload.get(MessagingConstants.EventDataKeys.Optimize.SCOPE);
-                if (rawScope == null || StringUtils.isNullOrEmpty(rawScope.toString())) {
-                    Log.warning(LOG_TAG, "%s - Unable to find a scope in the payload, payload will be discarded.", SELF_TAG);
-                    return;
-                }
-                final String decodedScopeString;
-                if (rawScope instanceof Variant) { // need to convert the scope Json depending on the source of the offers (optimize response event or previously cached offers)
-                    decodedScopeString = new String(encodingService.base64Decode(payload.get(MessagingConstants.EventDataKeys.Optimize.SCOPE).convertToString()));
-                } else {
-                    decodedScopeString = new String(encodingService.base64Decode((String) rawScope));
-                }
-                final JSONObject decisionScopeJson = new JSONObject(decodedScopeString);
-                decisionScope = (String) decisionScopeJson.get(MessagingConstants.EventDataKeys.Optimize.XDM_NAME);
-            } catch (final VariantException variantException) {
-                Log.warning(LOG_TAG, "%s - Exception occurred when converting a VariantMap to StringMap: %s", SELF_TAG, variantException.getMessage());
-                return;
-            } catch (final JSONException jsonException) {
-                Log.warning(LOG_TAG, "%s - Exception occurred when creating a JSON object from the encoded decision scope: %s", SELF_TAG, jsonException.getMessage());
-                return;
-            } catch (final NullPointerException nullPointerException) {
-                Log.warning(LOG_TAG, "%s - NullPointerException caught while trying to decode decision scope: %s", SELF_TAG, nullPointerException.getMessage());
-                return;
-            }
-
-            Log.debug(LOG_TAG, "%s - Offers IAM payload contained application identifier: (%s)", SELF_TAG, decisionScope);
-            if (!decisionScope.equals(offersConfig.applicationId)) {
-                Log.debug(LOG_TAG, "%s - ignoring Offers IAM payload, the retrieved application identifier did not match the expected application identifier: (%s).", SELF_TAG, offersConfig.applicationId);
-                return;
-            }
+        final String scope = (String) payload.get(SCOPE);
+        if (StringUtils.isNullOrEmpty(scope)) {
+            Log.warning(LOG_TAG, "%s - Unable to find a scope in the payload, payload will be discarded.", SELF_TAG);
+            return;
```

Comment:
take a look at how iOS handles this - we need to make sure this is a response for our request before we can even check for scope and payload contents:

https://github.com/adobe/aepsdk-messaging-ios/blob/inbound-changes/AEPMessaging/Sources/Messaging.swift#L161

Reply:
this is another thing i've fixed in the coming pr. I check that the request messages eventId is matching before checking for scope and payload.

Reply:
relevant code: https://github.com/rymorale/aepsdk-messaging-android/blob/iam-inbound-changes-proposition-interactions/code/messaging/src/phone/java/com/adobe/marketing/mobile/InAppNotificationHandler.java#L127-L129

Reply:
i'll move those changes into this pr for clarity.